src/providers/EtherspotContextProvider.tsx

```diff
@@ -230,11 +238,11 @@ const EtherspotContextProvider = ({
     assets: TokenListToken[],
     assetsChainId: number,
     balancesForAddress: string | null = accountAddress,
+    recompute: boolean = true,
   ) => {
     if (!sdk) return [];
-
     let computedAccount;
-    if (!balancesForAddress) {
+    if (!balancesForAddress && recompute) {
```

Comment:
or did you mean `|| recompute`?

Reply:
Actually it'll be && since by default recompute is true but in a specific case I'm passing it as false